Restructure documentation (#193)
xKDR/Survey.jl: docs/src/man/comparisons.md

# Comparison with other survey analysis tools

There are multiple alternatives that offer survey analysis tools, most notably

> [SAS/STAT](https://support.sas.com/rnd/app/stat/procedures/SurveyAnalysis.html)

Author: ```suggestion
[SAS](https://support.sas.com/rnd/app/stat/procedures/SurveyAnalysis.html),
[Stata](https://www.stata.com/features/survey-methods/)
```

> [SAS/STAT](https://support.sas.com/rnd/app/stat/procedures/SurveyAnalysis.html)
> and [R](https://CRAN.R-project.org/package=survey).

Reviewer: ```suggestion
[SAS](https://support.sas.com/rnd/app/stat/procedures/SurveyAnalysis.html), [STATA](https://www.stata.com/),
and [R](https://CRAN.R-project.org/package=survey).
```

Author: I added the link in the suggestion above, but I changed it to the link containing the survey analysis tool.

## R comparison

The inspiration for `Survey.jl` comes from R. Hence the syntax is in most cases
very similar to the syntax in the [`survey` package](https://cran.r-project.org/web/packages/survey/survey.pdf)

> from R. To showcase this we will use the `apisrs` dataset found in both R's

Reviewer: ```suggestion
from R. To showcase this we will use the `api` datasets found in both R's
```
You are showing other `api` datasets below

`survey` and `Survey.jl`. See the [Tutorial](@ref) section for more details about
the `api` datesets.

All examples show the R code first, followed by the Julia code.

#### Loading data

```R
data(api)
# all `api` datasets are loaded globally
```

```julia
srs = load_data("apisrs")
# only one dataset is loaded and stored in a variable
```

#### Creating a design

```R
srs = svydesign(id=~1, data=apisrs, weights=~pw) # simple random sample
strat = svydesign(id=~1, data=apistrat, strata=~stype, weights=~pw) # stratified
clus1 = svydesign(id=~dnum, data=apiclus1, weights=~pw) # clustered (one stage)
```

```julia
srs = SurveyDesign(apisrs; weights=:pw) # simple random sample
strat = SurveyDesign(apistrat; strata=:stype, weights=:pw) # stratified
clus1 = SurveyDesign(apiclus1; clusters=:dnum, weights=:pw) # clustered (one stage)
```

#### Creating a replicate design

```R

> bsrs = as.svrepdesign(srs, type="bootstrap")

Reviewer: ```suggestion
bsrs = as.svrepdesign(srs, type="subbootstrap")
```
`subbootstrap` is the Rao bootstrap that we are using.
